### PR TITLE
completed sample form with formix and yup [branch ch1755]

### PR DESCRIPTION
**Pull request by kburnell**

### CODE REVIEW

**Comment by csell5** [comment]
#### 📝Changed routes:


| [**/samples/form**](https://that-website-git-formiksample.that-conference.now.sh/samples/form) |
|:-:|
|<a href="https://that-website-git-formiksample.that-conference.now.sh/samples/form">Screenshot of /samples/form</a><br /><sup><a href="https://images.weserv.nl?url=microlink.nyc3.digitaloceanspaces.com/OxUY_nbwktArcsaUF2J6E08EKck_">(view full size)</a> |







Commit 8fe6de9a227e395f8b6207a8dd693e8d438cfe42 (https://that-website-git-formiksample.that-conference.now.sh).

File: components/shared/FormLayout.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+`;
+
+export const FormLabel = styled.label`
+  margin-bottom: 2rem;
+`;
+
+export const FormInput = styled.input`
+  display: block;
+  margin-top: 1rem;
+  border: 1px solid #d3d3d3;
```
Comment (saragibby) [inline components/shared/FormLayout.js]: be great to get these codes into `baseTheme`

File: components/shared/SquareButton.js [hunk 2]
```diff
@@ -3,23 +3,28 @@ import styled from 'styled-components';
...
-    onClick();
+    props.onClick();
   };
 
   return (
     <button
-      className={className}
-      onClick={() => clickTracking(`${icon} - ${iconClass}`)}
+      className={props.className}
+      color={props.color}
```
Comment (saragibby) [inline components/shared/SquareButton.js]: Love that you beefed this up to be more reusable 🤩

File: components/shared/FormLayout.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Grid } from 'styled-css-grid';
+
+import SquareButton from './SquareButton';
+
+export const FormGrid = styled(Grid)`
```
Comment (saragibby) [inline components/shared/FormLayout.js]: Really like how you created these shared layout items. Will really help the forms come together fast and all look consistent.

File: pages/samples/form.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  FormInputValidationMessage,
+} from '../../components/shared/FormLayout';
+
+const Main = styled.div``;
+
+const Heading = styled.h2`
+  margin-bottom: 1rem;
+`;
+
+const form = props => {
```
Comment (saragibby) [inline pages/samples/form.js]: Great idea to add this sample!

**Comment by kburnell** [comment]
@saragibby hex colors moved to baseTheme

File: components/shared/SquareButton.js [hunk 5]
```diff
@@ -3,23 +3,28 @@ import styled from 'styled-components';
 import Icon from './Icon';
 import * as gtag from '../../lib/gtag';
 
-const SquareButton = ({ className, icon, iconClass, onClick }) => {
+const SquareButton = props => {
```
Comment (GeekOnCoffee) [inline components/shared/SquareButton.js]: You can spread props at the end of a destructuring, so I generally like to leave this as `({ className, icon, iconClass, onClick, ...props })`

Then, down below, you can spread it back into your lower level component to handle passing anything else down.

**Pull request merged**